sds_data_manager/constructs/instrument_lambdas.py

```diff
@@ -47,7 +47,7 @@ def __init__(
         # TODO: if we need more variables change so we can pass as input
         lambda_environment = {
             "S3_BUCKET": f"{data_bucket.bucket_name}",
-            "S3_KEY_PATH": instrument_sources,
+            "S3_KEY_PATH": json.dumps(instrument_sources),
```

Comment:
Change the name to `S3_INSTRUMENT_SOURCE_PREFIXES` or something similar since now there could be more than one path. Is it actually prefixes, or do we need explicit individual keys for each source? If it is individual keys you could avoid the list objects and go to something like Object.load(): https://stackoverflow.com/questions/33842944/check-if-a-key-exists-in-a-bucket-in-s3-using-boto3